triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceConfig.scala

```diff
@@ -90,8 +92,24 @@ private[trigger] object ServiceConfig {
 
     opt[String]("auth")
       .optional()
-      .action((t, c) => c.copy(authUri = Some(Uri(t))))
-      .text("Auth middleware URI.")
+      .action((t, c) => c.copy(authBothUri = Some(Uri(t))))
+      .text(
+        "Sets both the internal and external auth URIs. Incompatible with --auth-internal and --auth-external."
+      )
+
+    opt[String]("auth-internal")
+      .optional()
+      .action((t, c) => c.copy(authInternalUri = Some(Uri(t))))
+      .text(
+        "Sets the internal auth URIs (used by the trigger service to connect directly to the middleware). Incompatible with --auth."
+      )
+
+    opt[String]("auth-external")
+      .optional()
+      .action((t, c) => c.copy(authExternalUri = Some(Uri(t))))
+      .text(
+        "Sets the external auth URI (the one returned to the browser). Incompatible with --auth."
```

Comment:
You could use `checkConfig` to ensure that `--auth` is not used in conjuction with `--auth-internal|external` directly in the options parser.

Reply:
Added checkConfig, which actually allowed me to add tests in `ServiceConfigTest` (I'd tried before, but with the check happening much later there wasn't anything to test at that level).

I'm not sure what test to _add_ on the middleware side, but I've made the internal and external URLs different so we can at least confirm that only the external one is used in produced URLs.